### PR TITLE
Feature/implement protobuf reader

### PR DESCRIPTION
**Pull request by scfitze**
* **Tickets addressed:** 241
* **Review:** By commit 
* **Merge strategy:** Merge (no squash) 

## Description

This feature should add a new C++ class ProtobufReader that can parse all of the binary data from a file and distribute this data during gameplay. The commits are organized by first adding the ProtobufReader and Implementation, then connecting this class to the ProtobufActor, and finally cleaning up extraneous files and UE Settings.

## Verification
Validated by viewing that the vizmessage contents were printed to the screen as expected during play game.

## Documentation
No Documentation needed

## Future work

Modifying or removing/replacing the ProtobufActor as we learn what the best methods are to distribute the data from the ProtobufReader to actors in the scene.

### CODE REVIEW

File: Source/cielim/Private/ProtobufActor.cpp [hunk 1]
```diff
@@ -18,38 +18,17 @@ AProtobufActor::AProtobufActor()
...
-    UE_LOG(LogTemp, Warning, TEXT("Success"));
+    this->protobufreader = new ProtobufReader("scenarioAsteroidArrival_UnityViz.bin");
+
 }
 
 // Called every frame
 void AProtobufActor::Tick(float DeltaTime)
 {
-	Super::Tick(DeltaTime);
+	Super::Tick(DeltaTime);
```
Comment (thibaudteil) [inline Source/cielim/Private/ProtobufActor.cpp]: This indent seems off from the rest

File: Source/cielim/Public/ProtobufReader.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+	bool get_eof() const { return eof; }
+
+private:
+	vizProtobufferMessage::VizMessage vizmessage;
+	std::fstream input;
+	google::protobuf::io::ZeroCopyInputStream* raw_input;
+	google::protobuf::io::CodedInputStream* coded_input;
+	bool eof;
+};
```
Comment (thibaudteil) [inline Source/cielim/Public/ProtobufReader.h:32]: Make sure to have an extra line at the end of files

File: Source/cielim/Public/ProtobufActor.h [hunk 3]
```diff
@@ -24,4 +25,7 @@ class CIELIM_API AProtobufActor : public AActor
 	// Called every frame
 	virtual void Tick(float DeltaTime) override;
 
+private:
+	ProtobufReader* protobufreader;
+	vizProtobufferMessage::VizMessage vizmessage;
 };
```
Comment (thibaudteil) [inline Source/cielim/Public/ProtobufActor.h:31]: Extra line again

File: Config/DefaultEngine.ini [hunk 4]
```diff
@@ -1,7 +1,8 @@
 
 
 [/Script/EngineSettings.GameMapsSettings]
-GameDefaultMap=/Engine/Maps/Templates/OpenWorld
+GameDefaultMap=/Game/Lvl_Visualization.Lvl_Visualization
+EditorStartupMap=/Game/Lvl_Visualization.Lvl_Visualization
 
 [/Script/WindowsTargetPlatform.WindowsTargetSettings]
 DefaultGraphicsRHI=DefaultGraphicsRHI_DX12
```
Comment (thibaudteil) [inline Config/DefaultEngine.ini:8]: Extra line

**Review by thibaudteil — APPROVED** [review]
Good progress

File: Source/cielim/Public/ProtobufReader.h (hunk 2 above)
Comment (scfitze) [reply Source/cielim/Public/ProtobufReader.h:32]: Strange my local files have the extra line but the gif status and diff is not picking it up as a change

Comment (scfitze) [reply Source/cielim/Public/ProtobufReader.h:32]: Ill try and add two lines and see what that does

File: Config/DefaultEngine.ini (hunk 4 above)
Comment (scfitze) [reply Config/DefaultEngine.ini:8]: This is another one of those config files that I don't directly edit, should I still try to add the new line?

Comment (thibaudteil) [reply Config/DefaultEngine.ini:8]: No worries in that case

**Comment by patkenneally** [comment]
Just to ensure you are aware, changes made to address PR comments will need to be rebased so that they occur in the commit where they originate. I'm happy to help with this.

File: Source/cielim/Private/ProtobufActor.cpp (hunk 1 above)
Comment (scfitze) [reply Source/cielim/Private/ProtobufActor.cpp]: Fixed

File: Source/cielim/Public/ProtobufReader.h (hunk 2 above)
Comment (scfitze) [reply Source/cielim/Public/ProtobufReader.h:32]: Fixed

File: Source/cielim/Public/ProtobufActor.h (hunk 3 above)
Comment (scfitze) [reply Source/cielim/Public/ProtobufActor.h:31]: Fixed

**Review by patkenneally — APPROVED** [review]

**Pull request merged**